### PR TITLE
Allow multiple derefs to be splitted in deref_separator

### PR DESCRIPTION
**Pull request by ouz-a**
Previously in #95649 only a single deref within projection was supported and multiple derefs caused a bunch of issues, this PR fixes those issues.

@oli-obk helped a ton again ❤️

### CODE REVIEW

File: compiler/rustc_mir_transform/src/deref_separator.rs [hunk 1]
```diff
@@ -23,14 +25,30 @@ pub fn deref_finder<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
...
-                            // temp value.
-                            let deref_place =
-                                Place::from(p_ref.local).project_deeper(p_ref.projection, tcx);
-                            patch.add_assign(
-                                loc,
-                                Place::from(temp),
-                                Rvalue::Use(Operand::Move(deref_place)),
-                            );
+                            // temp value, excluding projections we already covered.
+                            if idx == 1 {
```
Comment (oli-obk) [inline compiler/rustc_mir_transform/src/deref_separator.rs:29]: As far as I can tell the if and the else block are exactly the same now, so you can remove the condition and just keep one copy around

Comment (ouz-a) [reply compiler/rustc_mir_transform/src/deref_separator.rs:29]: Ah you are right

**Pull request merged**